android/quest/src/main/java/org/smartregister/fhircore/quest/util/extensions/ConfigExtensions.kt

```diff
@@ -164,11 +164,19 @@ fun List<ActionConfig>.handleClickEvent(
       ApplicationWorkflow.LAUNCH_INSIGHT_SCREEN ->
         navController.navigate(MainNavigationScreen.Insight.route)
       ApplicationWorkflow.DEVICE_TO_DEVICE_SYNC -> startP2PScreen(navController.context)
-      ApplicationWorkflow.LAUNCH_MAP ->
-        navController.navigate(
-          MainNavigationScreen.GeoWidgetLauncher.route,
-          bundleOf(NavigationArg.GEO_WIDGET_ID to actionConfig.id),
-        )
+      ApplicationWorkflow.LAUNCH_MAP -> {
+        val mapFragmentDestination = MainNavigationScreen.GeoWidgetLauncher.route
+        val isMapFragmentExists =
+          navController.currentBackStack.value.any { it.destination.id == mapFragmentDestination }
```

Comment:
`currentBackStack` is a restricted API and may be unstable. See the documentation here https://issuetracker.google.com/issues/217465473#comment6. If this can be allowed, we have to be sure this won't affect the app's functionality in any ways.

Reply:
Feedback resolved.